pytests/src/misc.rs

```diff
@@ -1,13 +1,23 @@
-use pyo3::prelude::*;
+use pyo3::{prelude::*, types::PyDict};
 
 #[pyfunction]
 fn issue_219() {
     // issue 219: acquiring GIL inside #[pyfunction] deadlocks.
     Python::with_gil(|_| {});
 }
 
+#[pyfunction]
+fn get_item_and_run_callback(dict: &PyDict, callback: &PyAny) -> PyResult<()> {
```

Comment:
I wonder what exactly we need to provide so that this could pass. The argument extraction code for supporting `Bound` argument and this could pass?

Reply:
Good question, it might already be possible to make this pass, I'll have a try...

Reply:
Indeed it turns out that just taking the `dict` and `callback` as `Bound<'_, PyDict>` and `Bound<'_, PyAny>` respectively fixes the crash (by changing the whole function body to use the new API).